tests/unit/scm/test_scm.py

```diff
@@ -0,0 +1,54 @@
+from dvc.repo.experiments import ExpRefInfo
+from dvc.scm import iter_revs
+
+
+def test_iter_revs(
+    tmp_dir,
+    scm,
+):
+    """
+    new         other
+     │            │
+    master (tag) ────┘
+     │
+    root
+    """
+    tmp_dir.scm_gen("foo", "init", commit="init")
+    rev_root = scm.get_rev()
+    tmp_dir.scm_gen("foo", "old", commit="old")
+    rev_old = scm.get_rev()
+    scm.checkout("new", create_new=True)
+    tmp_dir.scm_gen("foo", "new", commit="new")
+    rev_new = scm.get_rev()
+    scm.checkout("master")
```

Comment:
@skshetry related to #7005, we don't have `scm.default_branch` yet?

Reply:
Git does not have a concept of default branch locally. `init` uses `init.defaultBranch`/`--initial-branch` and defaults to `master`. `clone` uses currently active branch as initial branch by default.

 At least in scmrepo, it is not possible to know what branch is initial. So this feels like a tests-only requirement. Due to these issues, I have moved away from `default_branch` suggestion. I think the best way would be to keep a reference to `scm.active_branch()`, and use that instead.

Reply:
Here you may be able to avoid using hardcoded master by using `tmp_dir.branch` context manager.

https://github.com/iterative/dvc/blob/92f714e5ca7f4a9e7701a06f7d21b478049f4af7/dvc/testing/tmp_dir.py#L222